dask/array/core.py

```diff
@@ -3470,15 +3470,34 @@ def _vindex(x, *indexes):
     array([ 0,  9, 48,  7])
     """
     indexes = replace_ellipsis(x.ndim, indexes)
-    partial_slices = {i: ind for i, ind in enumerate(indexes)
-                      if isinstance(ind, slice) and ind != slice(None)}
+
+    num_indexes = []
```

Comment:
nit: Maybe `integer_indexes` would be a more descriptive name than `num_indexes`? "num" sounds a little more likely to include numpy arrays.

Reply:
Had `int_indexes` before. Also was debating `scalar_indexes`. No preferences amongst these personally. Please let me know what sounds best.

Reply:
`scalar_indexes` is probably clearest, that definitely excludes ND arrays :).

Reply:
Went with `nonfancy_indexes` given it now handles the scalar indices and partial slices. Happy to change the name if there are other thoughts.